ext/pdo/pdo_dbh.c

```diff
@@ -1139,8 +1154,14 @@ PHP_METHOD(PDO, quote)
 		Z_PARAM_LONG(paramtype)
 	ZEND_PARSE_PARAMETERS_END();
 
-	PDO_DBH_CLEAR_ERR();
 	PDO_CONSTRUCT_CHECK;
+
+	if (str_len == 0) {
```

Comment:
This looks extremely wrong. Quoting an empty string is totally normal and needs to work. Can you please add a test to make sure it continues working?